### PR TITLE
Maintain monorepo structure by merging reach-sh / reach-ide into this repository

### PR DESCRIPTION
**Pull request by hamirmahal**
I used https://stackoverflow.com/questions/1425892/how-do-you-merge-two-git-repositories/10548919#10548919 to attempt this.

[The other pull request I made to attempt this](https://github.com/reach-sh/reach-lang/pull/304) has faulty directories.

### CODE REVIEW

**Comment by jeapostrophe** [comment]
This is cool, I was expecting something more boring of just copying the files. Please educate me... how do we actually do releases to the VScode marketplace... is it possible to release from this single repo (like is there a file we generate?) or do installations copy the source repo itself (and thus it would be bad to make people download ALL of the Reach repo)?

**Comment by hamirmahal** [comment]
Thanks. I'm still looking into whether we can directly publish the Reach IDE extension from a subdirectory of our choosing. If that isn't viable, we should at least be able to [package the extension from subdirectory of our choosing, and "manually upload it to the Visual Studio Marketplace publisher management page"](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#publish-an-extension).

**Comment by hamirmahal** [comment]
Thanks. I'm still looking into whether we can directly publish the Reach IDE extension from a subdirectory of our choosing. If that isn't viable, we should at least be able to [package the extension from a subdirectory of our choosing, and "manually upload it to the Visual Studio Marketplace publisher management page"](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#publish-an-extension).

**Comment by hamirmahal** [comment]
Okay, I have some more information.

> how do we actually do releases to the VScode marketplace...

We need to [install `vsce` with `npm install -g vsce`, and then publish the extension using `vsce` with `vsce publish`](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#installation).


> is it possible to release from this single repo

Yes! I just tested this out a few seconds ago: https://marketplace.visualstudio.com/items?itemName=TestPublisher2021.extension-publish-test.

I was able to do this by running the above `vsce publish` command within my extension subdirectory.

My directory structure when I published this extension was and still is 
```
large-repo
--extension-publish-test (a folder with the extension's files)
--other-files.txt
```


> (like is there a file we generate?)

If we wanted to generate a file, and then manually drag and drop that file to publish the extension, I think we could do that using what I linked in [my earlier comment](https://github.com/reach-sh/reach-lang/pull/305#issuecomment-936486817), but that process doesn't seem as seamless and just running `vsce publish` from the extension's subdirectory.


> or do installations copy the source repo itself (and thus it would be bad to make people download ALL of the Reach repo)?

To my knowledge, installations will, at worst, copy a subset of all the files in the directory from which the developer ran `vsce publish`.

I think as long as we're running `vsce publish` from the _extension's_ subdirectory, we should be fine. The extension might not even publish correctly if we tried to run that command from this repository's top directory.

File: reach-ide/package.json [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	"displayName": "Reach IDE",
+	"description": "IDE for the Reach DApp programming language",
+	"author": "Reach Platform, Inc.",
+	"license": "EPL-2.0",
+	"version": "0.18.0",
+	"preview": true,
+	"icon": "images/reach-logo-dark.png",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/reach-sh/reach-ide"
```
Comment (jeapostrophe) [inline reach-ide/package.json:12]: This will change

File: reach-ide/package.json [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+	"categories": [
+		"Programming Languages",
+		"Snippets",
+		"Other"
+	],
+	"keywords": [
+		"reach",
+		"ethereum",
+		"algorand",
+		"blockchain"
```
Comment (jeapostrophe) [inline reach-ide/package.json:24]: conflux

**Comment by jeapostrophe** [comment]
And then I'll close the other repository

**Pull request merged**